eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Genesis.hs

```diff
@@ -112,165 +144,46 @@ instance FromCBOR AlonzoGenesis where
--- so even though we just have to pass the list to plutus, we still need to use the names of the parameters in order to sort the list.
--- In new versions, we want to represent the costmodel parameters directly as a list, so we can avoid this reordering, hence the name of this function.
-legacyParseCostModels :: Object -> Parser [(Language, CostModel)]
-legacyParseCostModels o =
-  do
-    plutusV1 <- o .:? "PlutusV1"
-    cms <- traverse (validateCostModel PlutusV1 . cmParamValues) plutusV1
-    pure $ maybeToList cms
-  where
-    cmParamValues :: Map Text Integer -> [Integer]
-    cmParamValues cmMap = mapMaybe (`Map.lookup` cmMap) plutusV1ParamNames
-
-parseCostModels :: Object -> Parser [(Language, CostModel)]
-parseCostModels o =
-  do
-    plutusV2 <- o .:? "PlutusV2"
-    maybeCostModels <- traverse (Aeson.withArray "PlutusV2 values" parseCostModelsV2) plutusV2
-    pure $ maybeToList maybeCostModels
-  where
-    parseCostModelsV2 :: Array -> Parser ((Language, CostModel))
-    parseCostModelsV2 array = do
-      paramValues <- mapM parseJSON $ Vector.toList array
-      validateCostModel PlutusV2 paramValues
-
-validateCostModel :: MonadFail m => Language -> [Integer] -> m (Language, CostModel)
-validateCostModel lang cmps = case mkCostModel lang cmps of
-  Left err -> fail $ show err
-  Right cm -> pure (lang, cm)
+          !> To agCoinsPerUTxOWord
+          !> To agCostModels
+          !> To agPrices
+          !> To agMaxTxExUnits
+          !> To agMaxBlockExUnits
+          !> To agMaxValSize
+          !> To agCollateralPercentage
+          !> To agMaxCollateralInputs
 
 instance FromJSON AlonzoGenesis where
   parseJSON = Aeson.withObject "Alonzo Genesis" $ \o -> do
-    coinsPerUTxOWord <- o .: "lovelacePerUTxOWord"
-    costmdls <- o .: "costModels"
-    prices <- o .: "executionPrices"
-    maxTxExUnits <- o .: "maxTxExUnits"
-    maxBlockExUnits <- o .: "maxBlockExUnits"
-    maxValSize <- o .: "maxValueSize"
-    collateralPercentage <- o .: "collateralPercentage"
-    maxCollateralInputs <- o .: "maxCollateralInputs"
+    agCoinsPerUTxOWord <- o .: "lovelacePerUTxOWord"
+    agCostModels <- o .: "costModels"
+    agPrices <- o .: "executionPrices"
+    agMaxTxExUnits <- o .: "maxTxExUnits"
+    agMaxBlockExUnits <- o .: "maxBlockExUnits"
+    agMaxValSize <- o .: "maxValueSize"
+    agCollateralPercentage <- o .: "collateralPercentage"
+    agMaxCollateralInputs <- o .: "maxCollateralInputs"
     return
       AlonzoGenesis
-        { coinsPerUTxOWord
-        , costmdls
-        , prices
-        , maxTxExUnits
-        , maxBlockExUnits
-        , maxValSize
-        , collateralPercentage
-        , maxCollateralInputs
+        { agCoinsPerUTxOWord
+        , agCostModels
+        , agPrices
+        , agMaxTxExUnits
+        , agMaxBlockExUnits
+        , agMaxValSize
+        , agCollateralPercentage
+        , agMaxCollateralInputs
```

Comment:
This could be done more compactly with sequential application:
```
instance FromJSON AlonzoGenesis where
  parseJSON = Aeson.withObject "Alonzo Genesis" $ \o ->
    AlonzoGenesis
      <$> o .: "lovelacePerUTxOWord"
      <*> o .: "costModels"
      <*> ...
```

Reply:
I don't like this style cause it is easy to mess up the order. For example if the order of fields has changed in the type itself, or I just happen to mess up the order in the implementation then compiler is not gonna help me in any way:

```diff
instance FromJSON AlonzoGenesis where
  parseJSON = Aeson.withObject "Alonzo Genesis" $ \o ->
    AlonzoGenesis
      <$> o .: "lovelacePerUTxOWord"
      <*> o .: "costModels"
      <*> ...
-      <*> o .: "agMaxTxExUnits"
-      <*> o .: "agMaxBlockExUnits"
+      <*> o .: "agMaxBlockExUnits"
+      <*> o .: "agMaxTxExUnits"
      <*> ...
```


That being said, I could use the record wildcards at the end, which would make just as compact as the version above, but still safe:
```haskell
    return AlonzoGenesis {..}
```

I'll do that when there will be any more suggestions from others

Reply:
Done. Thank you.